File: Serenity.Script.UI/Editor/AsyncLookupEditor.cs
using jQueryApi;
using System;
using System.Runtime.CompilerServices;

namespace Serenity
{
    [Imported(ObeysTypeSystem = true), Editor, OptionsType(typeof(LookupEditorOptions))]
    public class AsyncLookupEditor : LookupEditorBase<LookupEditorOptions, object>, IAsyncInit
    {
        public AsyncLookupEditor(jQueryObject hidden, LookupEditorOptions opt)
            : base(hidden, opt)
        {
        }

        protected override string GetLookupKey()
        {
            return options.LookupKey ?? base.GetLookupKey();
        }
    }
}

File: src/Serenity.Scripts/CoreLib/UI/Editors/AsyncLookupEditor.ts
import { Decorators } from "../../Decorators";
import { IGetEditValue, IReadOnly, ISetEditValue, IStringValue } from "../../Interfaces";
import { LookupEditorBase, LookupEditorOptions } from "./LookupEditor";

// legacy, don't use!
@Decorators.registerEditor('Serenity.AsyncLookupEditor',
    [ISetEditValue, IGetEditValue, IStringValue, IReadOnly])
export class AsyncLookupEditor extends LookupEditorBase<LookupEditorOptions, any> {
    constructor(hidden: JQuery, opt: LookupEditorOptions) {
        super(hidden, opt);
    }

    getLookupKey() {
        return (this.options.lookupKey ?? super.getLookupKey());
    }
}

